packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,344 @@
+  renderHref,
+  onPageChange,
+  startLabelText,
+  startAriaLabel,
+  endLabelText,
+  endAriaLabel,
+  totalPages,
+  ...rest
+}: PaginationProps): React.ReactElement {
+  const classes = classNames('ds-c-pagination', className);
+
+  /**
+   * `useState` and `useEffect` determine if
+   * mobile layout of component is rendered.
+   */
+
+  const [isMobile, setIsMobile] = React.useState(false);
+  React.useEffect(() => {
+    if (window) {
+      // Mobile media query derived from: https://design.cms.gov/guidelines/responsive/
+      const media = window.matchMedia('(max-width: 543px)');
+
+      if (media.matches !== isMobile) {
+        setIsMobile(media.matches);
+      }
+
+      const listener = () => {
+        setIsMobile(media.matches);
+      };
+
+      media.addEventListener('change', listener);
+      return () => media.removeEventListener('change', listener);
+    } else {
+      setIsMobile(true);
+    }
+  }, [isMobile]);
+
+  const pageChange = React.useCallback(
+    (page) => (evt: React.MouseEvent) => onPageChange(evt, page),
+    [onPageChange]
+  );
+
+  const pages = [];
+
+  /**
+   * If `compact` or `isMobile` is true,
+   * don't run code to populate `pages[]`.
+   */
+  if (!compact || !isMobile) {
+    const pageRange = paginationBuilder(currentPage, totalPages);
+
+    if (pageRange[0] >= 2) {
+      /**
+       * If `pageRange` begins with a page of 2 or greater,
+       * begin Pagination with Page 1
+       */
+      pages.push(
+        <Page
+          renderPageHref={renderHref(1)}
+          key="page-1"
+          index={1}
+          isActive={currentPage === 1}
+          onPageChange={pageChange(1)}
+        />
+      );
+
+      /**
+       * If `pageRange` doesn't equal 2, second Pagination element is Ellipses,
+       * otherwise page count continues.
+       */
+      if (pageRange[0] !== 2) {
+        pages.push(<Ellipses key="ellipses-1" />);
+      }
+    }
+
+    /**
+     * Renders all Page components in range (3 pages) to Pagination component.
+     */
+
+    pageRange.map((page) => {
+      pages.push(
+        <Page
+          renderPageHref={renderHref(page)}
+          key={`page-${page}`}
+          index={page}
+          isActive={currentPage === page}
+          onPageChange={pageChange(page)}
+        />
+      );
+    });
+
+    /**
+     * Defines if/when the Ellipses component appears
+     * at the end of the Pagination component -
+     * as long as there are fewer than 7 pages.
+     */
+    if (currentPage <= totalPages - 3 && totalPages > maxVisiblePages) {
+      if (currentPage < totalPages - 3) {
+        pages.push(<Ellipses key="ellipses-2" />);
+      }
+
+      pages.push(
+        <Page
+          renderPageHref={renderHref(totalPages)}
+          key={`page-${totalPages}`}
+          index={totalPages}
+          isActive={currentPage === totalPages}
+          onPageChange={pageChange(totalPages)}
+        />
+      );
+    }
+  }
+
+  const startIcon = (
```

Comment:
[musing] for both startIcon & endIcon, I wonder if these could be separated out into individual files, exported, and reused in other components.
Doesn't seem like something that needs to be done now, just a thought.

Reply:
agreed - i was kinda wishing we had an icon component in place for this because it would make the markup a little tighter